src/Domain/Block.cpp

```diff
@@ -28,6 +28,11 @@ Block<VolumeDim>::Block(
     : stationary_map_(std::move(stationary_map)),
       id_(id),
       neighbors_(std::move(neighbors)) {
+  ASSERT(external_boundary_conditions.empty() or
+             external_boundary_conditions.size() ==
+                 2 * VolumeDim - neighbors_.size(),
```

Comment:
This assert doesn't catch the case where `external_boundary_conditions` has the correct number of entries, but at the wrong directions. I suggest to test that, perhaps wrap the loop in `#ifdef SPECTRE_DEBUG` if necessary